src/os/bluestore/BitAllocator.cc

```diff
@@ -0,0 +1,1256 @@
+/*
+ * Bitmap based in-memory allocator.
+ * Author: Ramesh Chander, Ramesh.Chander@sandisk.com
+ */
+
+#include "BitAllocator.h"
+#include <assert.h>
+
+#define debug_assert assert
+#define MIN(x, y) ((x) > (y) ? (y) : (x))
+
+/*
+ * BmapEntityList functions.
+ */
+BmapEntityListIter::BmapEntityListIter(BmapEntityList *list) {
+	this->list = list;
+	this->start_idx = list->get_marker();
```

Comment:
I feel it makes it explicit and more readable. If not this we need some way of distinguish locals from class members. I can use any of those "m_*" ? or single underscore?